mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java

```diff
@@ -135,26 +139,37 @@ private void onInitializeRunningWorkerRequest(InitializeRunningWorkerRequestEven
     }
 
     private void onAssignedScheduleRequestEvent(AssignedScheduleRequestEvent event) {
+        TaskExecutorGateway gateway = null;
+        TaskExecutorRegistration info = null;
         try {
-            TaskExecutorGateway gateway =
-                resourceCluster.getTaskExecutorGateway(event.getTaskExecutorID()).join();
-
-            TaskExecutorRegistration info =
-                resourceCluster.getTaskExecutorInfo(event.getTaskExecutorID()).join();
+             gateway = resourceCluster.getTaskExecutorGateway(event.getTaskExecutorID()).join();
+             info = resourceCluster.getTaskExecutorInfo(event.getTaskExecutorID()).join();
+        } catch (Exception e) {
+            // we are not able to get the gateway, which either means the node is not great or some transient network issue
+            // we will retry the request
+            log.error(
+                "Failed to establish connection with the task executor {}; Resubmitting the request",
+                event.getTaskExecutorID(), e);
+            connectionFailures.increment();
+            self().tell(event.getScheduleRequestEvent().onFailure(e), self());
```

Comment:
nit: any backoff/delay needed to avoid short-circuiting the scheduler

Reply:
I'm sending a retry message here, which delays how it re-queues the original request.
```
log.error("Failed to submit the request {}; Retrying in {} because of ", event.getScheduleRequestEvent(), event.getThrowable());
            getTimers().startSingleTimer(
                getSchedulingQueueKeyFor(event.getScheduleRequestEvent().getRequest().getWorkerId()),
                event.onRetry(),
                intervalBetweenRetries);
```